From 3e23b5c31acc1759b1ed2aabd518964390f16c89 Mon Sep 17 00:00:00 2001
From: Maxim Uvarov <maxim.uvarov@linaro.org>
Date: Wed, 16 Apr 2014 18:15:29 +0400
Subject: [PATCH 1/2] Compile Snort as static library

Compile Snort as static library to use it's analyze functions
in other applications.
Signed-off-by: Maxim Uvarov <maxim.uvarov@linaro.org>
---
 src/Makefile.am |   40 +++++++++++++++++++++-------------------
 src/snort.c     |    8 ++------
 2 files changed, 23 insertions(+), 25 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index 1c1f826..f683dee 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -1,13 +1,13 @@
 ## $Id$
 AUTOMAKE_OPTIONS=foreign no-dependencies
 
-bin_PROGRAMS = snort
+noinst_LIBRARIES = libsnort.a
 
 if BUILD_SNPRINTF
 SNPRINTF_SOURCES = snprintf.c snprintf.h
 endif
 
-snort_SOURCES = cdefs.h \
+libsnort_a_SOURCES = cdefs.h \
 event.h \
 generators.h \
 sf_protocols.h \
@@ -62,20 +62,6 @@ rule_option_types.h \
 sfdaq.c sfdaq.h \
 idle_processing.c idle_processing.h idle_processing_funcs.h
 
-snort_LDADD = output-plugins/libspo.a \
-detection-plugins/libspd.a            \
-dynamic-plugins/libdynamic.a            \
-dynamic-output/plugins/liboutput.a      \
-preprocessors/libspp.a                \
-parser/libparser.a \
-target-based/libtarget_based.a \
-preprocessors/HttpInspect/libhttp_inspect.a \
-preprocessors/Stream5/libstream5.a \
-sfutil/libsfutil.a \
-control/libsfcontrol.a \
-file-process/libfileAPI.a \
-file-process/libs/libfile.a
-
 if BUILD_DYNAMIC_EXAMPLES
 EXAMPLES_DIR = dynamic-examples
 endif
@@ -85,9 +71,25 @@ SUBDIRS = sfutil win32 output-plugins detection-plugins dynamic-plugins preproce
 
 INCLUDES = @INCLUDES@
 
+libsnort_a_LIBADD = $(wildcard output-plugins/*.o) \
+$(wildcard detection-plugins/*.o)            \
+$(wildcard dynamic-plugins/*.o)            \
+$(wildcard dynamic-output/plugins/*.o)      \
+$(wildcard dynamic-output/*/*.o)      \
+$(wildcard preprocessors/*.o)                \
+$(wildcard parser/*.o) \
+$(wildcard target-based/*.o) \
+$(wildcard preprocessors/HttpInspect/*/*.o) \
+$(wildcard preprocessors/Stream5/*.o) \
+$(wildcard sfutil/*.o) \
+$(wildcard control/*.o) \
+$(wildcard file-process/*.o) \
+$(wildcard file-process/libs/*.o)
+
+
 if BUILD_SIDE_CHANNEL
-snort_LDADD += \
-side-channel/libsidechannel.a \
-side-channel/plugins/libsscm.a
+libsnort_a_LIBADD += \
+$(wildcard side-channel/*.o) \
+$(wildcard side-channel/plugins/*.o)
 SUBDIRS += side-channel
 endif
diff --git a/src/snort.c b/src/snort.c
index 0280107..ce5c52e 100644
--- a/src/snort.c
+++ b/src/snort.c
@@ -775,14 +775,9 @@ static int InlineFailOpen (void)
  * Returns: 0 => normal exit, 1 => exit on error
  *
  */
+#if 0
 int main(int argc, char *argv[])
 {
-	/* hack to add deps to -lrt for timer_create() and friends from odp */
-	static volatile int link_rt = 0;
-	timer_t timerid = 0;
-	if (link_rt)
-		timer_settime(timerid, 0, NULL, NULL);
-
 #if defined(WIN32) && defined(ENABLE_WIN32_SERVICE)
     /* Do some sanity checking, because some people seem to forget to
      * put spaces between their parameters
@@ -808,6 +803,7 @@ int main(int argc, char *argv[])
 
     return SnortMain(argc, argv);
 }
+#endif
 
 /*
  *
-- 
1.7.9.5

